frontend/src/features/public-form/components/SingleSubmissionModal/SingleSubmissionModal.tsx

```diff
@@ -0,0 +1,77 @@
+import { useNavigate } from 'react-router-dom'
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  Text,
+} from '@chakra-ui/react'
+
+import { FORM_SINGLE_SUBMISSION_VALIDATION_ERROR_MESSAGE } from '~shared/constants/'
+
+import { useIsMobile } from '~hooks/useIsMobile'
+import ButtonGroup from '~components/ButtonGroup'
+
+import { usePublicFormContext } from '~features/public-form/PublicFormContext'
+import { getPublicFormResponseSingpassLoginUrl } from '~features/public-form/utils/urls'
+
+interface SingleSubmissionModalProps {
+  formId: string
+  onClose: () => void
+  isOpen: boolean
+}
+
+// TODO: check if need to handle preview behavior
+export const SingleSubmissionModal = ({
+  formId,
+  isOpen,
+  onClose,
+}: SingleSubmissionModalProps) => {
+  const isMobile = useIsMobile()
+  const navigate = useNavigate()
+
+  const { handleLogout } = usePublicFormContext()
+
+  const singpassLogoutAndRedirectToFormLogin = () => {
+    if (!handleLogout) return
+    handleLogout()
+    navigate(getPublicFormResponseSingpassLoginUrl(formId))
```

Comment:
Is the intention here to reload the page? -- we also shouldn't change the url if we are trying to reload, the queryparams would have been lost.

Is the intention here to dismiss the modal? -- we can use `onClose()` to dismiss it.

Reply:
no, we are trying to reload the page instead. 

this function is invoked when the user clicks "Back to Singpass Login" button after the user gets a nric is already used modal. 

so in this case, we should redirect the user to the singpass page from scratch instead of dismissing the modal.

Reply:
im not sure if we want to keep the query params, but if we do, i can explore a way to bring along the query params (which im suspecting is used for form prefills) when navigating the route

Reply:
though this might not be necessary since this same user has already filled in the form according to this error message

Reply:
Understood. Sounds like we want to have something like a reload, and also agree that this specific usecase likely shouldn't retain the queryparams either.

Could I suggest naming it to be more generic such as `getPublicFormUrl(formId)`? This way, we can reuse the url builder in other situations.

Reply:
renamed